### PR TITLE
fix(rstest): hoist test API imports by source order

### PR DESCRIPTION
**Pull request by chenjiahan**
## Summary

Rstest mock hoisting previously recognized only the exact `@rstest/core` external fragment, so test APIs imported from `rstack/test`—either directly or through a user re-export—could be initialized after `rs.hoisted()` callbacks.

```js
import { rs } from 'rstack/test';

const mocks = rs.hoisted(() => ({
  mockedFn: rs.fn(),
}));
```

This PR tracks the test API import's source order and hoists the matching ESM fragment without hardcoding its package or external type, while preserving normal import ordering for non-hoisted methods.

## Related links

- [Rstack CLI](https://github.com/rstackjs/rstack-cli)
- [Rstack CLI migration](https://github.com/web-infra-dev/rspack/pull/14998)

## Checklist

- [x] Tests updated (or not required).
- [x] Documentation updated (or not required).

### CODE REVIEW

File: crates/rspack_plugin_rstest/src/mock_method_dependency.rs [hunk 1]
```diff
@@ -247,15 +266,15 @@ impl MockMethodDependencyTemplate {
     // Create a new fragment with higher priority (position=-3, before the mock
     // placeholder at -2 and importActual imports at -1) and insert at the beginning
     let content = conditional_fragment.content().to_string();
-    let rstest_import = ConditionalInitFragment::new(
+    let test_api_import = ConditionalInitFragment::new(
       content,
       InitFragmentStage::StageESMImports,
       -3,
```
[the remark this answers is not in the document]
Comment (chenjiahan) [reply crates/rspack_plugin_rstest/src/mock_method_dependency.rs:272]: Rstest expects a single test API provider, so we prefer not to add extra ordering logic for this edge case

File: crates/rspack_plugin_rstest/src/mock_method_dependency.rs [hunk 2]
```diff
@@ -247,15 +266,15 @@ impl MockMethodDependencyTemplate {
...
       target_key,
       None,
       RuntimeCondition::Boolean(true),
```
[the remark this answers is not in the document]
Comment (chenjiahan) [reply crates/rspack_plugin_rstest/src/mock_method_dependency.rs:275]: Test API providers are expected to be synchronous; supporting top-level-await wrappers is outside the scope of this fix.

**Review by 9aoy — APPROVED** [review]
